### PR TITLE
Skip TestSocketXDPGetInfo on EPERM

### PR DESCRIPTION
**Pull request by tklauser**
Skip `TestSocketXDPGetInfo` rather than failing it when the user lacks permissions to create AF_XDP sockets.

This allows to run the full test suite as an unprivileged user again.

### CODE REVIEW

File: socket_xdp_linux_test.go [hunk 1]
```diff
@@ -12,7 +13,9 @@ import (
 
 func TestSocketXDPGetInfo(t *testing.T) {
 	xdpsockfd, err := unix.Socket(unix.AF_XDP, unix.SOCK_RAW, 0)
-	if err != nil {
+	if errors.Is(err, unix.EPERM) {
```
Comment (aboch) [inline socket_xdp_linux_test.go:16]: I understand avoiding extra indentation, but I feel it is more readable is we test the error type only if the error is not nil. I am sure `error.Is` test handles the _nil_ use case, but I do not want to have to check.
```
if err != nil {
    if errors.Is(err, unix.EPERM) {
        t.Skipf("creating AF_XDP socket not permitted")
    }
     t.Fatal(err)
}
```

Yes, this is only test code, but quite often test code is looked upon to find examples.

**Comment by tklauser** [comment]
The test failure looks unrelated. It seems `TestRuleListFiltered` is somewhat flaky.

**Comment by aboch** [comment]
LGTM
